packages/wrangler/src/pages.tsx

```diff
@@ -670,252 +676,285 @@ async function buildFunctions({
+                try {
+                  return await assetsFetch(request);
+                } catch (thrown) {
+                  console.error(`Could not serve static asset: ${thrown}`);
+
+                  // TODO: Pretty error page
+                  return new Response(
+                    `[wrangler] Could not serve static asset: ${thrown}`,
+                    { status: 502 }
+                  );
+                }
               }
-            }
+            },
           },
-        },
 
-        kvPersist: true,
-        durableObjectsPersist: true,
-        cachePersist: true,
+          kvPersist: true,
+          durableObjectsPersist: true,
+          cachePersist: true,
 
-        ...miniflareArgs,
-      });
+          ...miniflareArgs,
+        });
 
-      // Wait for esbuild to finish building the script
-      while (!scriptReady) {}
+        // Wait for esbuild to finish building the script
+        while (!scriptReady) {}
 
-      try {
-        // `startServer` might throw if user code contains errors
-        const server = await miniflare.startServer();
-        console.log(`Serving at http://127.0.0.1:${port}/`);
+        try {
+          // `startServer` might throw if user code contains errors
+          const server = await miniflare.startServer();
+          console.log(`Serving at http://127.0.0.1:${port}/`);
 
-        if (process.env.BROWSER !== "none") {
-          await open(`http://127.0.0.1:${port}/`);
-        }
+          if (process.env.BROWSER !== "none") {
+            await open(`http://127.0.0.1:${port}/`);
+          }
 
-        EXIT_CALLBACKS.push(() => {
-          server.close();
-          miniflare.dispose().catch((err) => miniflare.log.error(err));
-        });
-      } catch (e) {
-        miniflare.log.error(e);
-        EXIT("Could not start Miniflare.", 1);
+          EXIT_CALLBACKS.push(() => {
+            server.close();
+            miniflare.dispose().catch((err) => miniflare.log.error(err));
+          });
+        } catch (e) {
+          miniflare.log.error(e);
+          EXIT("Could not start Miniflare.", 1);
+        }
       }
-    }
-  );
+    )
+    .command("functions", "Cloudflare Pages Functions", (yargs) =>
+      yargs.command(
+        "build",
+        "Compile a folder of Cloudflare Pages Functions into a single Worker",
+        (yargs) =>
+          yargs.options({
+            "script-path": {
+              type: "string",
+              default: "_worker.js",
+              description: "The location of the output Worker script",
+            },
+            minify: {
+              type: "boolean",
+              default: false,
+              description: "Minify the output Worker script",
+            },
+          }),
+        async ({ "script-path": scriptPath, minify }) => {
+          const functionsDirectory = "./functions";
+
+          await buildFunctions({
+            scriptPath,
+            functionsDirectory,
+            minify,
```

Comment:
Why not expose `sourcemap` and `watch` options too?

Certainly, I think one would want to generate sourcemaps if minifying, right? Currently sourcemaps will always default to false here.

Reply:
We don't need the `sourcemap` or `watch` options in our build image. We'll just be calling `wrangler pages functions build --script-path some/path/worker.js --minify`. If there's a use-case for either of these others being added, I'll happily add them, but until then, I'd rather minimize the contract.

Reply:
I see that for our own use case, these are not needed. Why offer minify as well then? If we are just uploading then why not always minify?

For the other use cases, I noted this in the changeset description: 

> ... or for people who want to do it themselves.

This implies that the build output might be used by others for some other reason??

Reply:
Nah, you're totally right. I've added the `sourcemap` option. Thanks for talking sense to me 😅